### PR TITLE
test(ticdc):  add run pulsar cluster in integration shell script

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: ref #10653

### What is changed and how it works?
 run pulsar cluster in integration shell script,  this script depends on the docker images file in https://github.com/pingcap/tiflow/pull/10640

to start a pulsar standalone cluster
```
run_pulsar_cluster $WORK_DIR normal
```

to start a mtls pulsar standalone cluster, the generated certs is in the specified working directory,  the file name is 
broker_client.cert.pem,  broker_client.key-pk8.pem and ca.cert.pem
```
run_pulsar_cluster $WORK_DIR mtls
```
to start a oauth2 enabled cluster pulsar standalone cluster
```
run_pulsar_cluster $WORK_DIR oauth
```

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by purelind** [comment]
wait for PR https://github.com/PingCAP-QE/ci/pull/2821  to be merged first.

**Review by asddongmen — APPROVED** [review]

File: tests/integration_tests/_utils/run_pulsar_cluster [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+)
+
+oauth_conf=$(
+	cat <<-EOF
+
+		authenticationEnabled=true
+		authenticationProviders=org.apache.pulsar.broker.authentication.AuthenticationProviderToken
+
+		brokerClientAuthenticationPlugin=org.apache.pulsar.client.impl.auth.oauth2.AuthenticationOAuth2
+		brokerClientAuthenticationParameters={"privateKey":"file://${workdir}/privateKey","audience":"https://dev-kt-aa9ne.us.auth0.com/api/v2/","issuerUrl":"https://dev-kt-aa9ne.us.auth0.com"}
```
Comment (CharlesCheung96) [inline tests/integration_tests/_utils/run_pulsar_cluster:64]: Will the auth0 dependency be replaced with an internal implementation in the future?

**Comment by CharlesCheung96** [comment]
Please describe the detailed functionality of the new script in the PR body.

File: tests/integration_tests/_utils/run_pulsar_cluster (hunk 1 above)
Comment (sdojjy) [reply tests/integration_tests/_utils/run_pulsar_cluster:64]: yes, I will start an oauth2 server in integration tests in next PR

File: tests/integration_tests/autorandom/run.sh [hunk 2]
```diff
@@ -21,7 +21,10 @@ function run() {
 	case $SINK_TYPE in
 	kafka) SINK_URI="kafka://127.0.0.1:9092/$TOPIC_NAME?protocol=open-protocol&partition-num=4&kafka-version=${KAFKA_VERSION}&max-message-bytes=10485760" ;;
 	storage) SINK_URI="file://$WORK_DIR/storage_test/$TOPIC_NAME?protocol=canal-json&enable-tidb-extension=true" ;;
-	pulsar) SINK_URI="pulsar://127.0.0.1:6650/$TOPIC_NAME?protocol=canal-json&enable-tidb-extension=true" ;;
+	pulsar)
+		run_pulsar_cluster $WORK_DIR normal
+		SINK_URI="pulsar://127.0.0.1:6650/$TOPIC_NAME?protocol=canal-json&enable-tidb-extension=true"
```
[the remark this answers is not in the document]
Comment (sdojjy) [reply tests/integration_tests/autorandom/run.sh:26]: tls is not enabled

File: tests/integration_tests/autorandom/run.sh [hunk 3]
```diff
@@ -21,7 +21,10 @@ function run() {
 	case $SINK_TYPE in
 	kafka) SINK_URI="kafka://127.0.0.1:9092/$TOPIC_NAME?protocol=open-protocol&partition-num=4&kafka-version=${KAFKA_VERSION}&max-message-bytes=10485760" ;;
 	storage) SINK_URI="file://$WORK_DIR/storage_test/$TOPIC_NAME?protocol=canal-json&enable-tidb-extension=true" ;;
-	pulsar) SINK_URI="pulsar://127.0.0.1:6650/$TOPIC_NAME?protocol=canal-json&enable-tidb-extension=true" ;;
+	pulsar)
+		run_pulsar_cluster $WORK_DIR normal
```
[the remark this answers is not in the document]
Comment (sdojjy) [reply tests/integration_tests/autorandom/run.sh:25]: no, only part of test case will be enabled

**Review by CharlesCheung96 — APPROVED** [review]

**Review by 3AceShowHand — APPROVED** [review]

**Pull request merged**